src/client/src/components/pages/participants/participantDates/index.jsx

```diff
@@ -4,81 +4,121 @@ import Header from "../../../abstract/header";
 
-export default class Date extends Component {
+class Date extends Component {
   state = {
     search: "",
-    rows: [
-      ["Participant Name", "Worker Name", "Start", "End", "Action"],
-      [
-        "Mohanned",
-        "mohannadhanafi",
-        "15/5/2017",
-        "15/12/2017",
-        <React.Fragment>
-          <i className="fas fa-trash-alt" />
-          <Link to="/participants/date/details">
-            <i className="fas fa-info-circle" />
-          </Link>
-        </React.Fragment>
-      ],
-      [
-        "Mohannad",
-        "mohannadhanafi",
-        "15/5/2017",
-        "15/12/2017",
-        <React.Fragment>
-          <i className="fas fa-trash-alt" />
-          <Link to="/participants/date/details">
-            <i className="fas fa-info-circle" />
-          </Link>
-        </React.Fragment>
-      ],
-      [
-        "Mohannad",
-        "mohannadhanafi",
-        "15/5/2017",
-        "15/12/2017",
-        <React.Fragment>
-          <i className="fas fa-trash-alt" />
-          <Link to="/participants/date/details">
-            <i className="fas fa-info-circle" />
-          </Link>
-        </React.Fragment>
-      ],
-      [
-        "Mohannad",
-        "mohannadhanafi",
-        "15/5/2017",
-        "15/12/2017",
-        <React.Fragment>
-          <i className="fas fa-trash-alt" />
-          <Link to="/participants/date/details">
-            <i className="fas fa-info-circle" />
-          </Link>
-        </React.Fragment>
-      ]
-    ]
+    rows: [],
+    message: ""
   };
+
   onChange = event => {
     const { value, name } = event.target;
     this.setState({ [name]: value });
   };
+
   goAdd = event => {
     this.props.history.push("/participants/date/add");
   };
+
   goBack = event => {
-    this.props.history.push("/participant/details");
+    const id = this.props.match.params.id;
+    this.props.history.push(`/participant/details/${id}`);
+  };
+
+  // axios to make requests from backend.. 
+  getDates = async () => {
+    const { dispatch } = this.props.context;
+    const id = this.props.match.params.id;
+    try {
+      const result = await axios(`/api/v2/participant/${id}/dates`);
+      const dates = result.data.participantDates;
+      let array = [["Participant Name/BB-No", "Worker Name/ID", "Start", "End", "Action"]];
+      if (dates.length === 0){
+        const msg = 'There is no dates yet !!';
+        array =[];          
+        this.setState({ message: msg, rows:array});
+      }
+      else{    
+      dates.map(row =>
+        array.push([
+          row.participant_id,
+          row.worker_id,
+          row.date_start.split("T")[0],
+          row.date_end.split("T")[0],
+          <>
+             <i className="fas fa-trash-alt"  onClick={() => this.onDelete(row.id)}/>
+             <Link to= {`/participant/${id}/date/details/${row.id}`}>
+               <i className="fas fa-info-circle" />
+             </Link>
+           </>
```

Comment:
I think it might be better if you just stored the dates / row data itself in state, instead of putting this JSX into state also.  Then, you could move this JSX into render() instead, looping over this.state.rows.  I think in general, rendered content is not supposed to go into this.state.  
It's easier for other people to find all of the JSX / html if it's in or near the render function, and it might be confusing if someone is trying to modify the display of this row and they don't realize they will find the rendered content inside getDates().